### ISSUE TITLE
"The long-running operation has failed" is an unclear message for customers

### ISSUE THREAD

**Issue by kazrael2119** [Azure/autorest.typescript#1791]
when sdk operation call was failing we were getting this info in the error response "The long-running operation has failed" and this message is not useful. Instead of it if we could get exact reason in the error message it will be very helpful.

![MicrosoftTeams-image](https://user-images.githubusercontent.com/98569699/227829970-7f3c402d-5a1a-46ca-8612-6763cc4aadbd.png)
"The long-running operation has failed" is an unclear message when we get this error. We can't know where the error comes from so that we don't know how to fix it.

Could we improve the experience for it?

**Comment by deyaaeldeen** [comment]
The design guidelines say there should be an `error` field that contains an error if the LRO status is `Failed`. I think we can update core-lro to print the content of that error field in this case. What do you think?

**Comment by kazrael2119** [comment]
> The design guidelines say there should be an `error` field that contains an error if the LRO status is `Failed`, see bullet 5 in [here](https://github.com/microsoft/api-guidelines/blob/vNext/azure/ConsiderationsForServiceDesign.md#long-running-operations). I think we can update core-lro to print the content of that error field in this case. What do you think?

Sounds great

**Comment by MaryGao** [comment]
@deyaaeldeen Agreed. In our recent experienc, when the provisioningState is failed the core-lro would throw the Exception directly. 

If we'd like to know what is the extract error message we have to dig deep into the response body via `onResponse` callback or debug into detailed payload. Like below payload:

```
{
  "id": "/subscriptions/49d64d54-e966-4c46-a868-1999802b762c/providers/Microsoft.Workloads/locations/EASTUS/operationStatuses/xxx",
  "name": "xxx",
  "resourceId": "/subscriptions/49d64d54-e966-4c46-a868-1999802b762c/resourceGroups/sdk-testing-vidya6_22_3_2023/providers/Microsoft.Workloads/monitors/haMonitor4/providerInstances/promethueusHa-provider",
  "status": "Failed",
  "startTime": "2023-03-23T10:49:28.6416273Z",
  "error": {
    "code": "InvalidProviderNameLength",
    "message": "\\n      The length of given Provider name: 'promethueusHa-provider' is not valid.\\n      Recommended Action: The valid length of provider name is minimum of 2 and maximum of 20 characters long.\\n    "
  }
}
```

The more useful information would be the error code and message.
